src/rgw/rgw_aws.cc

```diff
@@ -0,0 +1,259 @@
+#include "rgw_aws.h"
+#include <string>
+#include <iostream>
+#include <utility>
+#include "common/dout.h"
+#include <aws/core/Aws.h>
+#include <aws/sns/SNSClient.h>
+#include <aws/lambda/LambdaClient.h>
+#include <aws/lambda/model/InvokeRequest.h>
+#include <boost/lockfree/queue.hpp>
+#include <aws/core/utils/ARN.h>
+#include <aws/core/auth/AWSCredentials.h>
+#include <aws/sns/model/PublishRequest.h>
+
+#define dout_subsys ceph_subsys_rgw
+
+namespace rgw::aws {
+  static const int STATUS_OK = 1;
+  static const int STATUS_FULL = 2;
+  static const int STATUS_MANAGER_STOPPED = 3;
+  static const int STATUS_CLIENT_CLOSED = 4;
+
+  struct client_id_t {
+    const std::string accessKey;
+    const std::string accessSecret;
+    const std::string arn;
+    const bool useSSL;
+
+    client_id_t(const std::string accessKey, const std::string accessSecret,
```

Comment:
parameters could be `const std::string&`

Reply:
using ref causes it to throw some garbage value when called in publish_internal function

Reply:
I guess the memory is not valid later on.

Reply:
you pass reference into the function cal,, but copy into the data member.
also long as you don't do `std::move()` passing by reference should be ok.
the reference is not accessed after the ctor call ends.

Reply:
i still don't understand why the parameters cannot be `const std::string&`?